components/script/dom/performance.rs

```diff
@@ -93,13 +93,11 @@ impl PerformanceEntryList {
     pub fn clear_entries_by_name_and_type(
         &mut self,
         name: Option<DOMString>,
-        entry_type: Option<DOMString>,
+        entry_type: DOMString,
     ) {
         self.entries.retain(|e| {
-            name.as_ref().map_or(true, |name_| *e.name() != *name_) &&
-                entry_type
-                    .as_ref()
-                    .map_or(true, |type_| *e.entry_type() != *type_)
+            *e.entry_type() != *entry_type ||
+                name.as_ref().map_or(false, |name_| *e.name() != *name_)
```

Comment:
When the `name` is `None`, then the expectation is that all marks/measures are removed. If it's `Some`, then we need to compare the name and only remove entries that match.